tests/acceptance/stepDefinitions/publicLinkContext.js

```diff
@@ -5,48 +5,139 @@ const sharingHelper = require('../helpers/sharingHelper')
 const assert = require('assert')
 
 When(
-  'the user creates a new public link for file/folder/resource {string} using the webUI',
-  function (resource) {
-    return client.page.FilesPageElement
+  'the user creates/tries-to-create a new public link for file/folder/resource {string} using the webUI',
+  async function (resource) {
+    await client.page.FilesPageElement
       .filesList()
       .closeSidebar(100)
       .openPublicLinkDialog(resource)
+    return client.page.FilesPageElement.publicLinksDialog()
       .addNewLink()
   }
 )
 
 When(
-  'the user creates a new public link for file/folder/resource {string} using the webUI with',
-  function (resource, settingsTable) {
+  'the user creates/tries-to-create a new public link for file/folder/resource {string} using the webUI with',
+  async function (resource, settingsTable) {
     const settings = settingsTable.rowsHash()
-    return client.page.FilesPageElement
+    await client.page.FilesPageElement
       .filesList()
       .closeSidebar(100)
       .openPublicLinkDialog(resource)
-      .addNewLink(settings.role)
+    return client.page.FilesPageElement.publicLinksDialog().addNewLink(settings)
   }
 )
 
 When('the public uses the webUI to access the last public link created by user {string}', async function (linkCreator) {
-  const lastShareToken = await sharingHelper.fetchLastPublicLinkShare(linkCreator)
+  const lastShare = await sharingHelper.fetchLastPublicLinkShare(linkCreator)
+  const lastShareToken = lastShare.token
   return client.page.publicLinkFilesPage().navigateAndWaitTillLoaded(lastShareToken)
 })
 
-When('the public uses the webUI to access the last public link created by user {string} with password {string}', async function (linkCreator, password) {
-  const lastShareToken = await sharingHelper.fetchLastPublicLinkShare(linkCreator)
+When('last public share of the user {string} is deleted using API', async function (sharer) {
+  const lastShare = await sharingHelper.fetchLastPublicLinkShare(sharer)
+  const lastShareId = lastShare.id
+  return sharingHelper.deleteLastPublicShare(sharer, lastShareId)
+})
 
-  await client.page.publicLinkFilesPage().navigateAndWaitForPasswordPage(lastShareToken)
+Then('user {string} should not have any public link',
+  async function (sharer) {
+    const resp = await sharingHelper.getAllShares(sharer)
+    assert.deepStrictEqual(
+      resp, [], 'User has no empty shares. Response: ' + resp)
+  })
 
+Then('the fields of the last public link share response of user {string} should include',
+  function (linkCreator, dataTable) {
+    const fieldsData = dataTable.rowsHash()
+    return sharingHelper.assertUserLastPublicShareDetails(linkCreator, fieldsData)
+  })
+
+When('the public uses the webUI to access the last public link created by user {string} with password {string}', async function (linkCreator, password) {
```

Comment:
Looks similar to what we have already. Please refactor.

Reply:
Done :rocket: